### PR TITLE
Richard - Added Staff Form, Staff Form test, and Staff Form story

### LINKED ISSUE #9: Form Component for Staff plus tests
Create a React component for entering information about staff.
The user story this supports is:

As an instructor or admin
I can enter information about my course staff
So that I can delegate some functions to TAs/LAs

The acceptance criteria would be:

- [ ] The form has places to enter the GitHub id

- [ ] A  form called Staff form is visible
- [ ] There is a  test file to test staff form
- [ ] There is a story file for the staff form

### PR DESCRIPTION
**Pull request by richardfang888**
In this PR, we add a React component for a data entry form for the course staff, complete with a story and test file. As a user, I can now go onto the issue of making the Staff page accessible from the Courses data table and have the form be visible there.

Closes #9

### CODE REVIEW

File: frontend/src/main/components/Staff/StaffForm.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+          >
+            Cancel
+          </Button>
+        </Col>
+      </Row>
+    </Form>
+  );
+}
+
+export default StaffForm;
```
Comment (pconrad) [inline frontend/src/main/components/Staff/StaffForm.js]: I think the only thing we want on this form is literally GithubId and a submit button and literally nothing else.

The form will be embedded on a page where the course id is already there.   We'll use that information to send the add to the backend.

**Review by pconrad — CHANGES_REQUESTED** [review]
* Please see my comments on the team slack channel about PR descriptions and update this one accordingly.
* Please also update the issue to which this links by filling in the description (currently totally blank) and adding acceptance crtiera.
* Please also address the other concerns in the CR below.

File: frontend/src/main/components/Staff/StaffForm.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    formState: { errors },
+    handleSubmit,
+  } = useForm({ defaultValues: initialContents || {}, });
+
+  // Stryker restore all
+  const navigate = useNavigate();
+
+  // edit to add regex if needed; change Form register pattern and feedback to match
+  // const courseId_regex = '';
+  // const githubId_regex = '';
```
Comment (pconrad) [inline frontend/src/main/components/Staff/StaffForm.js]: It's not clear what the intentions of these comments are.  
Can we remove them?

**Review by pconrad — CHANGES_REQUESTED** [review]

**Review by pconrad — CHANGES_REQUESTED** [review]
Please review the comments on the team slack channel here:
https://ucsb-cs156-s24.slack.com/archives/C06SC553MLY/p1716870268080979

And see if this needs any reworking.

Adding staff should always be done in the context of a course.

Someone is a TA or an LA *for a specific course*. They are not an LA or TA "in general".

So the form for adding a staff member needs to have the information about the course right on the page, so that we know which course we are adding a TA or LA to.

I would suggest that the URL should contain the id of the course and the information about that course should be displayed at the top of the page.

I would suggest, actually not even HAVING a separate page for adding staff. Instead, it should go on the course show page.   And there is no need for edit.  Only add and delete.

**Comment by richardfang888** [comment]
> Please review the comments on the team slack channel here: https://ucsb-cs156-s24.slack.com/archives/C06SC553MLY/p1716870268080979
> 
> And see if this needs any reworking.
> 
> Adding staff should always be done in the context of a course.
> 
> Someone is a TA or an LA _for a specific course_. They are not an LA or TA "in general".
> 
> So the form for adding a staff member needs to have the information about the course right on the page, so that we know which course we are adding a TA or LA to.
> 
> I would suggest that the URL should contain the id of the course and the information about that course should be displayed at the top of the page.
> 
> I would suggest, actually not even HAVING a separate page for adding staff. Instead, it should go on the course show page. And there is no need for edit. Only add and delete.

I think I have addressed these changes in my most recent commit by making courseId implicit in the form and displayed alongside the id. I have adjusted the tests for the form as well.

**Review by abhishekambastha04 — APPROVED** [review]
LGTM

**Review by NevMan1 — APPROVED** [review]
LGTM

**Review by andrewpengucsb — APPROVED** [review]
LGTM

**Pull request merged**